### PR TITLE
Updating the plan files

### PR DESCRIPTION
**Pull request by johnmccrae**
## Description
Hab Plans for both Windows and Linux are blowing up in the verify pipeline, correcting them here. We are explicitly setting the version of bundler to 2.3.7 as we have discovered a bug in later versions (>= 2.3.26) where bundler does not properly detect the platform when run under Habitat and that was leading to FFI installation failures. That problem can be overcome in 2 ways - 1) Falling back to 2.3.7 or 2) Installing the gem before running bundler, setting the bundle configuration option for specific_platform, and then when calling bundle install we add the "--prefer_local" switch so when bundler hits FFI, it sees the local gem already installed and keeps going.

#gemlock_major_upgrade

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] If `Gemfile.lock` has changed, I have used `--conservative` to do it and included the full output in the Description above.
- [ ] All new and existing tests passed.
- [ ] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -224,10 +224,7 @@ GEM
     faraday-net_http (3.0.2)
     fauxhai-ng (9.3.0)
       net-ssh
-    ffi (1.16.3)
-    ffi (1.16.3-x64-mingw-ucrt)
-    ffi (1.16.3-x64-mingw32)
-    ffi (1.16.3-x86-mingw32)
+    ffi (1.17.0)
```
Comment (tpowell-progress) [inline Gemfile.lock:227]: Need to make sure all build channels are ok with the upgrade. It should be harmless, but `ffi` has caused nebulous problems in the past.

File: habitat/plan.ps1 [hunk 2]
```diff
@@ -91,6 +91,24 @@ function Invoke-Build {
 
         $env:_BUNDLER_WINDOWS_DLLS_COPIED = "1"
 
+        # Write-BuildLine " ** What is my pkg_prefix set to?"
+        # Write-BuildLine $pkg_prefix
+
+        # Write-BuildLine " ** FFI needs ltmain.sh, running libtoolize to create it"
+        # libtoolize
```
Comment (tpowell-progress) [inline habitat/plan.ps1:98]: Clean out unused code prior to merge.

File: habitat/plan.sh [hunk 3]
```diff
@@ -41,6 +41,8 @@ do_before() {
 }
 
 do_download() {
+  build_line "Setting up the safe directory for the build"
+  git config --global --add safe.directory /src
```
Comment (tpowell-progress) [inline habitat/plan.sh:45]: I'm experiencing a need for this in Windows as well, but not sure if your git path edit is fixing it.

**Review by tpowell-progress — DISMISSED** [review]
Cleanup and validate need for `safe.directory` in Windows as well (for Docker build)

File: Gemfile.lock (hunk 1 above)
Comment (johnmccrae) [reply Gemfile.lock:227]: agreed

**Review by neha-p6 — CHANGES_REQUESTED** [review]

File: Gemfile.lock (hunk 1 above)
Comment (johnmccrae) [reply Gemfile.lock]: I added ffi 1.16 back into the gemfile.lock. I didn't want to rock the boat too much here.

File: habitat/plan.ps1 [hunk 4]
```diff
@@ -68,13 +71,12 @@ function Invoke-Prepare {
 @"%~dp0ruby.exe" "%~dpn0" %*
 "@
         $gem_file | Set-Content "$PWD\\gem.bat"
-        $env:Path += ";c:\\Program Files\\Git\\bin"
+        $env:Path += ";c:\\Program Files\\Git\\bin;"
         $env:HAB_BLDR_CHANNEL = "LTS-2024"
         $env:HAB_STUDIO_SECRET_NODE_OPTIONS = "--dns-result-order=ipv4first"
         $env:HAB_STUDIO_SECRET_HAB_BLDR_CHANNEL = "LTS-2024"
         $env:HAB_STUDIO_SECRET_HAB_FALLBACK_CHANNEL= "LTS-2024"
```
Comment (neha-p6) [inline habitat/plan.ps1:78]: are these environment variables needed now that we are setting it above. Specifically `$env:HAB_BLDR_CHANNEL` is getting set in the beginning of this file.

**Review by neha-p6 — CHANGES_REQUESTED** [review]

**Review by neha-p6 — APPROVED** [review]

**Review by neha-p6 — APPROVED** [review]

**Pull request merged**